pandas/core/arrays/numpy_.py

```diff
@@ -0,0 +1,446 @@
+import numbers
+
+import numpy as np
+
+from pandas._libs import lib
+from pandas.compat.numpy import function as nv
+
+from pandas.core.dtypes.common import extract_array
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+from pandas.core.dtypes.inference import is_list_like
+
+from pandas import compat
+from pandas.core import nanops
+
+from .base import ExtensionArray, ExtensionOpsMixin
+
+
+class PandasDtype(ExtensionDtype):
+    """
+    A Pandas ExtensionDtype for NumPy dtypes.
+
+    .. versionadded:: 0.24.0
+
+    This is mostly for internal compatibility, and is not especially
+    useful on its own.
+
+    Parameters
+    ----------
+    dtype : numpy.dtype
+    """
+    _metadata = ('_dtype',)
+
+    def __init__(self, dtype):
+        dtype = np.dtype(dtype)
+        self._dtype = dtype
+        self._name = dtype.name
+        self._type = dtype.type
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def _is_numeric(self):
```

Comment:
do we need to deal with boolean here?

Reply:
Bool is considered numeric by `DataFrame._get_numeric_dtypes`